File: erpnext_import_tracker/public/js/purchase_order.js
frappe.ui.form.on("Purchase Order", {
	refresh(frm) {
		if (frm.doc.docstatus !== 1 || !frm.doc.custom_is_import) {
			return;
		}

		frm.add_custom_button(
			__("Import Shipment"),
			() => {
				frappe.call({
					method: "erpnext_import_tracker.import_tracker.doctype.import_shipment.import_shipment.make_import_shipment",
					args: { purchase_order: frm.doc.name },
					freeze: true,
					freeze_message: __("Creating Import Shipment..."),
					callback: (r) => {
						if (r.message) {
							frappe.set_route("Form", "Import Shipment", r.message);
						}
					},
				});
			},
			__("Create")
		);
	},
});

File: erpnext_import_tracker/events/purchase_order.py
import frappe

from erpnext_import_tracker.import_tracker.doctype.import_shipment.import_shipment import (
	create_import_shipment,
)


def on_submit(doc, method=None):
	"""Auto-create an Import Shipment when an import Purchase Order is submitted."""
	if not doc.get("custom_is_import"):
		return

	if frappe.db.exists("Import Shipment", {"purchase_order": doc.name}):
		return

	name = create_import_shipment(doc.name)
	frappe.msgprint(
		frappe.utils.get_link_to_form("Import Shipment", name),
		title="Import Shipment created",
		indicator="green",
	)
